packages/financial-templates-lib/src/price-feed/TraderMadePriceFeed.js

```diff
@@ -0,0 +1,283 @@
+const { PriceFeedInterface } = require("./PriceFeedInterface");
+const { parseFixed } = require("@uma/common");
+const moment = require("moment");
+
+// An implementation of PriceFeedInterface that uses TraderMade api to retrieve prices.
+class TraderMadePriceFeed extends PriceFeedInterface {
+  /**
+   * @notice Constructs the TraderMadePriceFeed.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} web3 Provider from truffle instance to connect to Ethereum network.
+   * @param {String} pair Representation of the pair the price feed is tracking.
+   * @param {String} apiKey TraderMade Data API key.
+   * @param {Integer} botLookback How far in the past the historical prices will be available using getHistoricalPrice.
+   * @param {Integer} affiliateLookback How far in the past the historical prices will be available using getHistoricalPricePeriods.
+   * @param {Object} networker Used to send the API requests.
+   * @param {Function} getTime Returns the current time.
+   * @param {Integer} minTimeBetweenUpdates Min number of seconds between updates. If update() is called again before
+   *      this number of seconds has passed, it will be a no-op.
+   * @param {Number} priceFeedDecimals Number of priceFeedDecimals to use to convert price to wei.
+   * @param {Number} ohlcPeriod Number of minutes interval between ohlc prices requested from TraderMade.
+   */
+  constructor(
+    logger,
+    web3,
+    apiKey,
+    pair,
+    botLookback,
+    affiliateLookback,
+    networker,
+    getTime,
+    minTimeBetweenUpdates,
+    priceFeedDecimals = 18,
+    ohlcPeriod = 10 // Only 5, 10, 15 minutes is supported by TraderMade.
+  ) {
+    super();
+    this.logger = logger;
+    this.web3 = web3;
+
+    this.apiKey = apiKey;
+    this.pair = pair;
+
+    this.botLookback = botLookback;
+    this.affiliateLookback = affiliateLookback;
+    this.networker = networker;
+    this.getTime = getTime;
+    this.minTimeBetweenUpdates = minTimeBetweenUpdates;
+
+    this.toBN = this.web3.utils.toBN;
+
+    this.priceFeedDecimals = priceFeedDecimals;
+    this.ohlcPeriod = ohlcPeriod;
+
+    this.convertPriceFeedDecimals = number => {
+      // Converts price result to wei
+      // returns price conversion to correct decimals as a big number
+      return this.toBN(parseFixed(number.toString(), priceFeedDecimals).toString());
+    };
+  }
+
+  getCurrentPrice() {
+    return this.currentPrice;
+  }
+
+  getHistoricalPrice(time, verbose = false) {
```

Comment:
This should be an `async` function now to match the `PriceFeedInterface`. Additionally, instead of returning `null` this function should throw an Error. The [CryptoWatchPriceFeed](https://github.com/UMAprotocol/protocol/blob/master/packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.js#L69) is probably the best model that you can follow for help!

Reply:
fixed